### ISSUE TITLE
Regression between Docker container v1.29.18 and v1.29.20

### ISSUE THREAD

**Issue by cedi** [tailscale/tailscale#5549]
### What is the issue?

When updating my Kubernetes Sidecar deployment from Version v1.29.18 to v1.29.20 I can no longer establish a connection to the Tailnet.

Background: I inject tailscale as a sidecar to an application, top make the application reachable via tailnet. Therefore I patch my Application using Kustomize with the following patch:

```yaml
- op: add
  path: "/spec/template/spec/containers/-"
  value:
    name: ts-sidecar
    imagePullPolicy: Always
    image: ghcr.io/tailscale/tailscale:v1.29.20
    env:
      - name: TS_KUBE_SECRET
        value: repo-server-tailscale-auth
      - name: TS_USERSPACE
        value: "false"
      - name: TS_AUTH_KEY
        valueFrom:
          secretKeyRef:
            name: repo-server-tailscale-auth
            key: AUTH_KEY
            optional: true
    securityContext:
      capabilities:
        add:
          - NET_ADMIN
          - SYS_MODULE

```

This works perfectly with v1.29.18, but when I upgrade to v1.29.20 and above, the tailscaled logs show a connection-reset-by-peer happening, right after connection:

```plain
/tailscale/run.sh: set: line 7: can't access tty; job control turned off
Starting tailscaled
Running tailscale up
2022/09/06 12:04:42 logtail started
2022/09/06 12:04:42 Program starting: v1.29.20-t9514ed33d, Go 1.18.4-ts149f7d88f1: []string{"tailscaled", "--socket=/tmp/tailscaled.sock", "--state=kube:repo-server-tailscale-auth", "--statedir=/tmp"}
2022/09/06 12:04:42 LogID: dca67b01a862f99ad7c8ec68e652757242ee872ba3c1ef6779caf74d603ab422
2022/09/06 12:04:42 logpolicy: using system state directory "/var/lib/tailscale"
logpolicy.ConfigFromFile /var/lib/tailscale/tailscaled.log.conf: open /var/lib/tailscale/tailscaled.log.conf: no such file or directory
logpolicy.Config.Validate for /var/lib/tailscale/tailscaled.log.conf: config is nil
2022/09/06 12:04:42 wgengine.NewUserspaceEngine(tun "tailscale0") ...
2022/09/06 12:04:42 setting link attributes: netlink receive: no such file or directory
2022/09/06 12:04:42 router: v6nat = false
2022/09/06 12:04:42 dns: [rc=unknown ret=direct]
2022/09/06 12:04:42 dns: using *dns.directManager
2022/09/06 12:04:42 link state: interfaces.State{defaultRoute=eth0 ifs={eth0:[10.244.3.116/32]} v4=true v6=false}
2022/09/06 12:04:42 magicsock: disco key = d:8bc673ee9bd50516
2022/09/06 12:04:42 Creating WireGuard device...
2022/09/06 12:04:42 Bringing WireGuard device up...
2022/09/06 12:04:42 Bringing router up...
2022/09/06 12:04:42 external route: up
2022/09/06 12:04:42 Clearing router settings...
2022/09/06 12:04:42 Starting link monitor...
2022/09/06 12:04:42 Engine created.
2022/09/06 12:04:42 monitor: ip rule deleted: {Family:2 DstLength:0 SrcLength:0 Tos:0 Table:254 Protocol:0 Scope:0 Type:1 Flags:0 Attributes:{Dst:<nil> Src:<nil> Gateway:<nil> OutIface:0 Priority:5210 Table:254 Mark:4294967295 Pref:<nil> Expires:<nil> Metrics:<nil> Multipath:[]}}
2022/09/06 12:04:42 monitor: ip rule deleted: {Family:2 DstLength:0 SrcLength:0 Tos:0 Table:253 Protocol:0 Scope:0 Type:1 Flags:0 Attributes:{Dst:<nil> Src:<nil> Gateway:<nil> OutIface:0 Priority:5230 Table:253 Mark:4294967295 Pref:<nil> Expires:<nil> Metrics:<nil> Multipath:[]}}
2022/09/06 12:04:42 monitor: ip rule deleted: {Family:2 DstLength:0 SrcLength:0 Tos:0 Table:0 Protocol:0 Scope:0 Type:7 Flags:0 Attributes:{Dst:<nil> Src:<nil> Gateway:<nil> OutIface:0 Priority:5250 Table:0 Mark:4294967295 Pref:<nil> Expires:<nil> Metrics:<nil> Multipath:[]}}
2022/09/06 12:04:42 monitor: ip rule deleted: {Family:2 DstLength:0 SrcLength:0 Tos:0 Table:52 Protocol:0 Scope:0 Type:1 Flags:0 Attributes:{Dst:<nil> Src:<nil> Gateway:<nil> OutIface:0 Priority:5270 Table:52 Mark:0 Pref:<nil> Expires:<nil> Metrics:<nil> Multipath:[]}}
2022/09/06 12:04:42 Start
2022/09/06 12:04:42 using backend prefs for "_daemon": Prefs{ra=false dns=false want=true routes=[] nf=on Persist{lm=, o=[MYvTN], n=[huw5C] u="cedi"}}
2022/09/06 12:04:42 Backend: logs: be:dca67b01a862f99ad7c8ec68e652757242ee872ba3c1ef6779caf74d603ab422 fe:
2022/09/06 12:04:42 control: client.Login(false, 0)
2022/09/06 12:04:42 control: doLogin(regen=false, hasUrl=false)
2022/09/06 12:04:42 health("overall"): error: not in map poll
2022/09/06 12:04:42 control: control server key from https://controlplane.tailscale.com: ts2021=[fSeS+], legacy=[nlFWp]
2022/09/06 12:04:42 control: RegisterReq: onode=[MYvTN] node=[huw5C] fup=false
2022/09/06 12:04:42 control: RegisterReq: got response; nodeKeyExpired=false, machineAuthorized=true; authURL=false
2022/09/06 12:04:42 ipnserver: conn2: connection from userid 0; root has access
2022/09/06 12:04:42 Start
2022/09/06 12:04:42 control: client.Shutdown()
2022/09/06 12:04:42 control: client.Shutdown: inSendStatus=0
2022/09/06 12:04:42 control: mapRoutine: quit
2022/09/06 12:04:42 control: Client.Shutdown done.
2022/09/06 12:04:42 using backend prefs for "_daemon": Prefs{ra=false dns=false want=true routes=[] nf=on Persist{lm=, o=[MYvTN], n=[huw5C] u="cedi"}}
2022/09/06 12:04:42 router: somebody (likely systemd-networkd) deleted ip rules; restoring Tailscale's
2022/09/06 12:04:42 Backend: logs: be:dca67b01a862f99ad7c8ec68e652757242ee872ba3c1ef6779caf74d603ab422 fe:
2022/09/06 12:04:42 control: client.Login(false, 0)
2022/09/06 12:04:42 control: doLogin(regen=false, hasUrl=false)
2022/09/06 12:04:42 control: control server key from https://controlplane.tailscale.com: ts2021=[fSeS+], legacy=[nlFWp]
2022/09/06 12:04:42 control: RegisterReq: onode=[MYvTN] node=[huw5C] fup=false
2022/09/06 12:04:42 control: RegisterReq: got response; nodeKeyExpired=false, machineAuthorized=true; authURL=false
2022/09/06 12:04:59 active login: argocd-repo-server-6cb6cfb899-qwcnk.cedi.github
2022/09/06 12:04:59 Switching ipn state NoState -> Starting (WantRunning=true, nm=true)
2022/09/06 12:04:59 magicsock: SetPrivateKey called (init)
2022/09/06 12:04:59 wgengine: Reconfig: configuring userspace WireGuard config (with 0/7 peers)
2022/09/06 12:04:59 wgengine: Reconfig: configuring router
2022/09/06 12:04:59 wgengine: Reconfig: configuring DNS
2022/09/06 12:04:59 dns: Set: {DefaultResolvers:[] Routes:{} SearchDomains:[] Hosts:8}
2022/09/06 12:04:59 dns: Resolvercfg: {Routes:{} Hosts:8 LocalDomains:[]}
2022/09/06 12:04:59 dns: OScfg: {Nameservers:[] SearchDomains:[] MatchDomains:[]}
2022/09/06 12:04:59 peerapi: serving on http://100.102.73.175:42165
2022/09/06 12:04:59 peerapi: serving on http://[fd7a:115c:a1e0:efe3::6466:49af]:42165
2022/09/06 12:04:59 magicsock: home is now derp-4 (fra)
2022/09/06 12:04:59 magicsock: adding connection to derp-4 for home-keep-alive
2022/09/06 12:04:59 control: NetInfo: NetInfo{varies=true hairpin=false ipv6=false udp=true derp=#4 portmap= link=""}
2022/09/06 12:04:59 magicsock: 1 active derp conns: derp-4=cr0s,wr0s
2022/09/06 12:04:59 Switching ipn state Starting -> Running (WantRunning=true, nm=true)
2022/09/06 12:04:59 derphttp.Client.Connect: connecting to derp-4 (fra)
2022/09/06 12:04:59 ipnserver: conn2: ReadMsg: read unix /tmp/tailscaled.sock->@: read: connection reset by peer
2022/09/06 12:04:59 magicsock: endpoints changed: 94.130.187.160:41554 (stun), 10.244.3.116:43543 (local)
/tailscale/run.sh: fg: line 85: job (null) not created under job control
```

The docker containers show up in the Tailscale Admin Panel just fine, but they are marked as "Not connected".

The v1.28.18 logs (using the same TS_AUTH_KEY) look like this:

```plain
Starting tailscaled
Running tailscale up
2022/09/06 11:43:58 logtail started
2022/09/06 11:43:58 Program starting: v1.29.18-t3e06b9ea7, Go 1.18.4-ts149f7d88f1: []string{"tailscaled", "--socket=/tmp/tailscaled.sock", "--state=kube:repo-server-tailscale-auth", "--statedir=/tmp"}
2022/09/06 11:43:58 LogID: 5732748cd9c031f7c06f7348e7beb58364c9d141bccfd73ae71b090f26600c43
2022/09/06 11:43:58 logpolicy: using system state directory "/var/lib/tailscale"
logpolicy.ConfigFromFile /var/lib/tailscale/tailscaled.log.conf: open /var/lib/tailscale/tailscaled.log.conf: no such file or directory
logpolicy.Config.Validate for /var/lib/tailscale/tailscaled.log.conf: config is nil
2022/09/06 11:43:58 wgengine.NewUserspaceEngine(tun "tailscale0") ...
2022/09/06 11:43:58 setting link attributes: netlink receive: no such file or directory
2022/09/06 11:43:58 router: v6nat = false
2022/09/06 11:43:58 dns: [rc=unknown ret=direct]
2022/09/06 11:43:58 dns: using *dns.directManager
2022/09/06 11:43:58 link state: interfaces.State{defaultRoute=eth0 ifs={eth0:[10.244.3.97/32]} v4=true v6=false}
2022/09/06 11:43:58 magicsock: disco key = d:e2aa1666152f2d55
2022/09/06 11:43:58 Creating WireGuard device...
2022/09/06 11:43:58 Bringing WireGuard device up...
2022/09/06 11:43:58 Bringing router up...
2022/09/06 11:43:58 external route: up
2022/09/06 11:43:58 Clearing router settings...
2022/09/06 11:43:58 Starting link monitor...
2022/09/06 11:43:58 Engine created.
2022/09/06 11:43:58 Start
2022/09/06 11:43:58 using backend prefs for "_daemon": Prefs{ra=false dns=false want=true routes=[] nf=on Persist{lm=, o=[dirmO], n=[QLWYL] u="cedi"}}
2022/09/06 11:43:58 Backend: logs: be:5732748cd9c031f7c06f7348e7beb58364c9d141bccfd73ae71b090f26600c43 fe:
2022/09/06 11:43:58 control: client.Login(false, 0)
2022/09/06 11:43:58 control: doLogin(regen=false, hasUrl=false)
2022/09/06 11:43:58 health("overall"): error: not in map poll
2022/09/06 11:43:58 control: control server key from https://controlplane.tailscale.com: ts2021=[fSeS+], legacy=[nlFWp]
2022/09/06 11:43:58 control: RegisterReq: onode=[dirmO] node=[QLWYL] fup=false
```

### Steps to reproduce

Deploy a sidecar container in Kubernetes in version v1.29.20 with a `TS_AUTH_KEY` that is marked `ephemeral` **and** `reusable` .
Reusable for obvious reasons, Ephemeral, since the pods originate from Kubernetes Deployments/Replicasets.

_No response_

### Are there any recent changes that introduced the issue?

v1.29.20 is the first Docker-Container that does not work anymore

### OS

Linux, Other

### OS version

docker tailscale:v1.29.20

### Tailscale version

v1.29.20

**Comment by hollie** [comment]
I can replicate this with a basic docker setup (docker-compose file with nginx and tailscale).

v1.28 connects
v1.29 and v1.30 do not with the same error as @cedi mentioned above.

My test docker-compose file looks like this:

```
version: '2.4'

services:
    demo:
        hostname: demo
        image: nginx:alpine                               # Your image goes here
        expose:
            - 80
        cpus: 2
        mem_limit: 4096m
        restart: on-failure

    tailscale:
        # https://rnorth.org/tailscale-docker/
        # $ docker-compose up
        # $ docker-compose exec tailscale tailscale up
        #    To authenticate, visit: https://login.tailscale.com/a/SOME_HEX_CODE
        image: tailscale/tailscale:v1.29
        cap_add:
            - NET_ADMIN
            # - SYS_MODULE             # usually not needed, depends on host OS
        network_mode: 'service:demo'
        volumes:
            - /dev/net/tun:/dev/net/tun
            - ./data/tailscale:/var/lib/tailscale
        cpus: 2
        mem_limit: 4096m
        restart: on-failure
```

**Comment by bradfitz** [comment]
cc @maisem

**Comment by GingerGeek** [comment]
I've also come across this, #5543 fixes it, would it be possible to get new container images with it backported?

**Comment by GingerGeek** [comment]
It looks like a new tailscale release addresses this

https://github.com/tailscale/tailscale/releases/tag/v1.30.1

**Comment by hollie** [comment]
Verified that the 1.30.1 image is working as expected. Thanks for the update!

**Comment by DentonGentry** [comment]
Fixed in 1.30.1
